FormalConjectures/Wikipedia/Grimm.lean

```diff
@@ -0,0 +1,41 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Grimm's conjecture
+
+*Reference:* [Wikipedia](https://en.wikipedia.org/wiki/Grimm%27s_conjecture)
+-/
+
+def IsSetOfPrimes (k : ℕ+) (ps : Fin k → ℕ) : Prop :=
+  (∀ i : Fin k, Prime (ps i)) ∧ (∀ (i : Fin k) (j : Fin k), ps i = ps j → i = j)
```

Comment:
There's [Function.Embedding](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Logic/Embedding/Basic.html#Function.Embedding) to express injective functions.